Orange/widgets/gui.py

```diff
@@ -62,6 +62,10 @@ def __init__(self, parent=None, **kwargs):
         v.setVisible(False)
         v.setResizeMode(QHeaderView.ResizeToContents)
 
+    def setModel(self, model):
+        super().setModel(model)
+        model.view = self
```

Comment:
This solution is incorrect. A single data model can be attached to multiple views.

Did you have a look at QSortFilterProxyModel?